Add render json to health

francesco-loreti opened the pull request:
### Motivation / Background
This Pull Request has been created because the health controller currently responds only in HTML, and I needed to add the JSON render for an API-only application.

### Detail

This Pull Request changes HealthController

### Additional information

<!-- Provide additional information such as benchmarks, references to other repositories, or alternative solutions. -->

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

francesco-loreti committed:
Add render json to health

Edouard-chin commented on railties/lib/rails/health_controller.rb:
```diff
@@ -18,10 +18,10 @@ module Rails
   #
   #   Rails.application.routes.draw do
-  #     get "healthz" => "rails/health#show", as: :rails_health_check
+  #     get "health" => "rails/health#show", as: :rails_health_check
```
This is a convention and not a typo.

guilleiguaran commented on actionpack/CHANGELOG.md:
```diff
@@ -1,3 +1,7 @@
+*   Add `render json` into `HealthController`.
```
We can improve this a bit, perhaps by describing the purpose of the changes instead of pointing out directly what changes were made in the code.

JuanVqz committed:
Add JSON support to health controller

The health controller now responds to JSON requests, providing a structured response with status and timestamp information. This enhancement facilitates easier integration with monitoring tools and load balancers.
Additionally, tests have been added to verify the JSON responses for both success and error scenarios.
